app/scripts/migrations/077.test.js

```diff
@@ -319,4 +413,868 @@ describe('migration #77', () => {
+            },
+          },
+        });
+      });
+    });
+
+    describe('state transformation to ahead of migration 86', () => {
+      it('should delete `provider` from the NetworkController state, if the user already has `providerConfig` in NetworkController state, without interferring with the rest of the migration', async () => {
+        const oldStorage = {
+          meta: {
+            version: 76,
+          },
+          data: {
+            TokenListController: {
+              tokenList: {
+                '0x514910771af9ca656af840dff83e8264ecf986ca': {
+                  address: '0x514910771af9ca656af840dff83e8264ecf986ca',
+                  symbol: 'LINK',
+                  decimals: 18,
+                },
+              },
+              tokensChainsCache: {
+                1: {
+                  timestamp: 1234,
+                  data: [
+                    {
+                      address: '0x514910771af9ca656af840dff83e8264ecf986ca',
+                      symbol: 'LINK',
+                      decimals: 18,
+                    },
+                    {
+                      address: '0xc00e94cb662c3520282e6f5717214004a7f26888',
+                      symbol: 'COMP',
+                      decimals: 18,
+                    },
+                  ],
+                },
+              },
+            },
+            NetworkController: {
+              provider: { foo: 'bar ' },
+              providerConfig: { fizz: 'buzz' },
+            },
+          },
+        };
+        const newStorage = await migration77.migrate(oldStorage);
+        expect(newStorage).toStrictEqual({
+          meta: {
+            version: 77,
+          },
+          data: {
+            TokenListController: {
+              tokenList: {
+                '0x514910771af9ca656af840dff83e8264ecf986ca': {
+                  address: '0x514910771af9ca656af840dff83e8264ecf986ca',
+                  symbol: 'LINK',
+                  decimals: 18,
+                },
+              },
+              tokensChainsCache: {
+                1: {
+                  timestamp: 1234,
+                  data: {
+                    '0x514910771af9ca656af840dff83e8264ecf986ca': {
+                      address: '0x514910771af9ca656af840dff83e8264ecf986ca',
+                      symbol: 'LINK',
+                      decimals: 18,
+                    },
+                    '0xc00e94cb662c3520282e6f5717214004a7f26888': {
+                      address: '0xc00e94cb662c3520282e6f5717214004a7f26888',
+                      symbol: 'COMP',
+                      decimals: 18,
+                    },
+                  },
+                },
+              },
+            },
+            NetworkController: {
+              providerConfig: { fizz: 'buzz' },
+            },
+          },
+        });
+      });
+
+      it('should not delete `network` from the NetworkController state, if there is no `networkId` in NetworkController state', async () => {
```

Comment:
Test description fix:
```suggestion
      it('should not delete `provider` from the NetworkController state, if there is no `providerConfig` in NetworkController state', async () => {
```